charts/tidb-cluster/templates/scripts/_start_tidb.sh.tpl

```diff
@@ -34,6 +34,7 @@ then
     ARGS="${ARGS} --enable-binlog=true"
 fi
 
+SLOW_LOG_FILE=${SLOW_LOG_FILE:-""}
 if [[ ! -z "${SLOW_LOG_FILE}" ]]
 then
     ARGS="${ARGS} --log-slow-query=${SLOW_LOG_FILE:-}"
```

Comment:
Is it necessary to have `${SLOW_LOG_FILE:-}` after setting it on line 37 and checking it on line 38?

Reply:
On line 38 it will be undefined. The purpose of line 37 is just to define it and set a default value if it is not defined. It is a little verbose and obscure as bash is, but I think this idiom is easy to get used to and understand whereas the other ones are trickier. And overall it is much safer to use these techniques and have undefined variables be an error.

Reply:
Oh, so I think you are more directly asking about `:-`. That allows for the variable to be unset and not throw an undefined variable error.

```
${parameter:-word}
    If parameter is unset or null, the expansion of word is substituted. 
    Otherwise, the value of parameter is substituted.
```